06-building-a-program-start-to-finish/writing-tests/output/inverted-index.spec.js

```diff
@@ -0,0 +1,184 @@
+const books = require('../books.json');
+const adventureBooks = require('../adventure-books.json');
+const emptyArray = require('../emptyBook.json');
+const invalidContent = require('../invalid-content.json');
+const invalidFile = require('../invalid-file.json');
+const invalidKey = require('../invalid-key.json');
+
+const invertedIndex = new InvertedIndex();
+
+describe('Validate File ', () => {
+  it(`should verify file have keys named "title" and "text"
+    with string for values`, () => {
+    expect(invertedIndex.validateFile(invalidContent))
+      .toBe('Invalid file content');
+    expect(invertedIndex.validateFile(invalidKey))
+      .toBe('Invalid file content');
+  });
+  it('should verify file is not empty', () => {
+    expect(invertedIndex.validateFile(emptyArray))
+      .toBe('Empty file');
+  });
+  it('should verify a file is valid', () => {
+    expect(invertedIndex.validateFile(books))
+      .toBe('Valid file');
+  });
+  it('should verify a file is invalid', () => {
+    expect(invertedIndex.validateFile(invalidFile))
+      .toBe('Invalid file');
+  });
+});
+
+describe('Tokenize function ', () => {
+  const bookToTokenize = [{ title: 'Alice , / ?', text: 'enters a a.' }];
+
+  it('should return an " array " of words for file content input', () => {
+    expect(Array.isArray(InvertedIndex
+      .tokenize(`${bookToTokenize[0].title} ${bookToTokenize[0].text}`)))
+      .toBe(true);
+  });
+  it(`should return an "array" of filtered words for a 
+  file content input`, () => {
+    expect(InvertedIndex
+      .tokenize(`${bookToTokenize[0].title} ${bookToTokenize[0].text}`))
+      .toEqual(
+        ['alice', 'enters', 'a']
+      );
+  });
+});
+
+describe('Create index for a JSON file', () => {
+  beforeAll(() => {
+    invertedIndex.createIndex(books, 'books.json');
+    invertedIndex.createIndex(adventureBooks, 'adventure-books.json');
+  });
+  it('should generate the correct index', () => {
+    expect(invertedIndex.getIndex('books.json').and)
+      .toEqual(
+        [0, 1]
+      );
+    expect(invertedIndex.getIndex('adventure-books.json').bringing)
+      .toEqual(
+        [1]
+      );
+  });
+});
+
+describe('Get index of a JSON file', () => {
+  it('should return "undefined" if index does not exist', () => {
+    expect(invertedIndex.getIndex(' '))
+      .toEqual(undefined);
+    expect(invertedIndex.getIndex('books4'))
+      .toEqual(undefined);
+  });
+  it(`should verify that the index of JSON file can be gotten 
+  after file indexed`, () => {
+    expect(invertedIndex.getIndex('books.json'))
+      .toBeDefined();
+  });
+  it(`should map the string keys to the correct objects in 
+  the JSON array`, () => {
+    expect(invertedIndex.getIndex('adventure-books.json'))
```

Comment:
I feel like you should have tested this with your createIndex, because you're checking to see if the index created is accurate.